### ISSUE TITLE
Index method on PCardList returns None

### ISSUE THREAD

**Issue by BDTrooper** [EskoSalaka/mtgtools#24]
Hi,

if i have a PCardList, lets call it "proxies", then

proxies.index(proxies[0])
returns None.

proxies.pop(0) works as expected (where i would expect that somehow the index method is involved).

Maybe it is because you put a PCard object directly into self._cards.index(card), thus into the persistent package?

**Comment by EskoSalaka** [comment]
Hmmm strange, for me (running Python 3.10) it does seem to work as expected. For example the following gives me correct results:

```
counterspells = cards.where_exactly(name='Counterspell')
werebears = cards.where_exactly(name='Werebear')
counterspells.pprint()
werebears.pprint()

print(counterspells.index(counterspells[5])) # prints 5
print(counterspells.index(werebears[0])) # throws a ValueError: Werebear (wc02) is not in list
```

The _cards of PCardList is a PersistentList of PCard objects. PersistentList should work the same way as a normal python List, so as far as i understand, the self._cards.index(card) should also work the same way. 

Maybe in a certain situation something is messed up with the _cards of a PCardList in such a way that the index-method is unable to work properly. Im not sure how it works under the hood, but perhaps some required comparison (like PCard.__eq__) doesnt work correctly.

I can think of a few reasons:
- maybe python version have some differences
-  bug in PCardList in which the _cards attribute of the list get somehow messed up.  Maybe after some specific operation like __add__, __sub__ or similar  it becomes broken
- maybe in some cases, coming from the database, PCards or other objects are not somehow correctly initialized/populated. PersistentList should work the same way as a normal python List, but maybe there are some limitations in some cases
- maybe there are some broken PCard objects in the database

Do you also get a ValueError or just None, i think this method is supposed to raise a ValuError if the object cant be found. How did you create the proxies-list and did you perform any operations on it?

**Comment by BDTrooper** [comment]
I'm using 3.11.7

I have streamlined my example, i'm using a scryfall sourced DB created with mtgtools:

```
from pathlib import Path, PurePath
from mtgtools.MtgDB import MtgDB
mtg_data = Path(os.getenv('mtg_data'))
mtgpool = MtgDB(mtg_data.joinpath('mtgpool.fs').as_posix())
mtgpool.scryfall_bulk_update()
```
 
Then i use it in another session

```
mtg_data = Path(os.getenv('mtg_data'))
mtg_decks = Path(os.getenv('mtg_decks'))
mtgpool = MtgDB(mtg_data.joinpath('mtgpool.fs').as_posix())
mtgcards = mtgpool.root.scryfall_cards
len(mtgcards)
```
95992

```
mtgcards.where_exactly(name = 'Bane of Progress').where_exactly(set = 'cma')[0]
```
Bane of Progress (cma)

```
if mtgcards.index(mtgcards.where_exactly(name = 'Bane of Progress').where_exactly(set = 'cma')[0]) is None:
    print('Its a bug!')
```
Its a bug!

**Comment by EskoSalaka** [comment]
Thanks for the extra info and the explanation. What i posted earlier was incorrect!

I think i see my mistake now. I wrote some debug prints in the index-method of the PCardList and I think my copilot  "accidentally" fixed the bug that was in it. Thats why when i ran that script of mine, the result came out as expected. I only realized it when i did a fresh install and looked at the method again...

The bug is in the index-method, it is missing the return statement. It does find the card (does not raise an exception) but returns None

```
    def index(self, card):
        """Returns the index where the given card object is located in the list.

        Args:
            card (PCard): The card object to be searched.
        """
        self._cards.index(card)
```
